core/src/surplus.ts

```diff
@@ -0,0 +1,95 @@
+import type { SurplusAuction, SurplusAuctionUnexistant, SurplusAuctionBaseData } from './types';
+import { getEarliestDate } from './helpers/getEarliestDate';
+import BigNumber from './bignumber';
+import getContract from './contracts';
+import { Contract } from 'ethers';
+
+const surplusAuctionLastIndex = async (contract: Contract) => {
+    const auctionsQuantityBinary = await contract.kicks();
+    return new BigNumber(auctionsQuantityBinary._hex);
+};
+
+const getAuctionState = (earliestEndDate: Date, greatestBid: number) => {
+    const isBidExpired = new Date() > earliestEndDate;
+    const haveBids = greatestBid === 0;
+    if (haveBids) {
+        if (isBidExpired) {
+            return 'ready-for-collection';
+        }
+        return 'have-bids';
+    }
+    if (isBidExpired) {
+        return 'requires-restart';
+    }
+    return 'just-started';
+};
+
+export const fetchSurplusAuctionByIndex = async function (
+    network: string,
+    auctionIndex: number,
+    contractOrNull: Contract | null = null
+): Promise<SurplusAuction | SurplusAuctionUnexistant> {
+    let contract: Contract;
+    if (!!contractOrNull) {
+        contract = contractOrNull;
+    } else {
+        contract = await getContract(network, 'MCD_FLAP');
+    }
+
+    const auctionData = await contract.bids(auctionIndex);
+    const isAuctionDeleted = new BigNumber(auctionData.end).eq(0);
+    const baseAuctionInfo: SurplusAuctionBaseData = {
+        network,
+        id: auctionIndex,
+    };
+
+    if (isAuctionDeleted) {
+        const auctionLastIndex = await surplusAuctionLastIndex(contract);
+        if (auctionLastIndex.lt(auctionIndex)) {
+            throw new Error('No active auction found with this id');
```

Comment:
I would make it clear that auction with this id never existed in the first place, not just not found?